### PR TITLE
FOGL-6956: S2OPCUA user timestamp compatibility with OPCUA south

### PR DESCRIPTION
**Pull request by Innovative-ashwin**
Signed-off-by: Innovative-ashwin <sinha.ashwini111@gmail.com>

### CODE REVIEW

**Review by MarkRiddoch — DISMISSED** [review]
Looks good to me

File: include/opcua.h [hunk 1]
```diff
@@ -122,6 +123,7 @@ class OPCUA
         std::string            	m_clientPrivate;
         std::string            	m_certAuth;
         std::string            	m_caCrl;
+	struct timeval		m_userts;
```
Comment (RVerhoeff) [inline include/opcua.h:126]: The OPCUA class represents a client connection to an OPC UA server. It is not the right place to put a member that is used to construct a single Reading.

File: opcua.cpp [hunk 2]
```diff
@@ -285,7 +287,18 @@ DatapointValue* dpv = NULL;
...
+                        SOPC_DateTime srcTimestamp = value->SourceTimestamp;
+                        time_t seconds;
+                        if (SOPC_Time_ToTimeT(srcTimestamp, &seconds) == SOPC_STATUS_OK)
+                        {
+                       		double TimeAsSecondsFloat = ((double) srcTimestamp) / 1.0E7;         // divide by 100 nanoseconds
+                                double integerPart = 0.0;
+                                m_userts.tv_sec = seconds;
+                                m_userts.tv_usec = (suseconds_t) (1E6 * modf(TimeAsSecondsFloat, &integerPart));
+                        }
 			ingest(points, 0);
```
Comment (RVerhoeff) [inline opcua.cpp:300]: This is the only place where _OPCUA::ingest_ is called so it is easy to change its definition. It's second argument is currently "_long user_ts_" which is passed as zero so it is never used. In its implementation, it sets the Reading's user timestamp using the OPCUA class member _m_userts_. Instead, change the argument list of _OPCUA::ingest_ to _(vector<Datapoint *> points, timeval &user_ts)_. Create "_timeval userts_" as a local variable, do the math as you do now, and then pass _userts_ directly to _OPCUA::ingest_.

**Review by RVerhoeff — CHANGES_REQUESTED** [review]
The definition and use of _timeval userts_ should be addressed before merging.

File: opcua.cpp (hunk 2 above)
Comment (Innovative-ashwin) [reply opcua.cpp]: fixed

File: include/opcua.h (hunk 1 above)
Comment (Innovative-ashwin) [reply include/opcua.h]: fixed

**Review by RVerhoeff — APPROVED** [review]
Great! Thanks.

**Pull request merged**